apps/fhir/bluebutton/views/generic.py

```diff
@@ -91,10 +92,18 @@ def get(self, request, resource_type, *args, **kwargs):
 
     def fetch_data(self, request, resource_type, *args, **kwargs):
         resource_router = get_resourcerouter(request.crosswalk)
+        api_ver = 'v2' if request.path.startswith("/v2") else 'v1'
+        # BB2-291 v2 switch enforced here, entry of all fhir resources queries
+        if request.path.startswith("/v2"):
+            if (not waffle.switch_is_active('bfd_v2')):
+                raise exceptions.NotFound("bfd_v2 disabled.")
+            if (not waffle.flag_is_active(request, 'bfd_v2_flag')):
+                raise exceptions.NotFound("bfd_v2_flag not active.")
+
```

Comment:
Example continued...

The `self.version` instance variable could be used here for version logic:

```suggestion
        # BB2-291 v2 switch enforced here, entry of all fhir resources queries
        if self.version == 2:
            if (not waffle.switch_is_active('bfd_v2')):
                raise exceptions.NotFound("bfd_v2 disabled.")
            if (not waffle.flag_is_active(request, 'bfd_v2_flag')):
                raise exceptions.NotFound("bfd_v2_flag not active.")

```